main/src/main/scala/sbt/Project.scala

```diff
@@ -479,7 +479,15 @@ object Project extends ProjectExtra {
     previousOnUnload(s.runExitHooks())
   }
 
-  def setProject(session: SessionSettings, structure: BuildStructure, s: State): State = {
+  def setProject(session: SessionSettings, structure: BuildStructure, s: State): State =
+    setProject(session, structure, s, st => st)
```

Comment:
This is fine but could be `identity` instead of `st => st`. I sometimes like to use `identity` for clarity even though they are functionally the same.